airbyte-commons-server/src/main/java/io/airbyte/commons/server/scheduler/SynchronousResponse.java

```diff
@@ -61,7 +61,8 @@ public static <T, U> SynchronousResponse<T> fromTemporalResponse(final TemporalR
         configId,
         jobOutput != null ? jobOutput.getConnectorConfigurationUpdated() : false,
         createdAt,
-        endedAt);
+        endedAt,
+        jobOutput != null ? jobOutput.getFailureReason() : null);
```

Comment:
Since we're using a few things from the jobOutput now, I wonder if it makes sense just to pass the jobOutput through and let metadata pull things from it as needed

Reply:
That makes sense to me, i'll try it out

Reply:
Any idea why we're passing `jobOutput` and `responseOutput` separately to this method, when they both came from the `temporalResponse` in the first place? That might be a bit of cleanup as well

Reply:
@pedroslopez lmk what you think of the result. I think it looks cleaner - a little less DRY [here](https://github.com/airbytehq/airbyte-platform/pull/144/files#diff-754353c954321b0adc883a661e79b4ae6e961e3965a0059ab26e691647697751R56-R57) but with a bit of a refactor of `track()` in [this issue ](https://app.zenhub.com/workspaces/connector-operations-6318ede5eda32e0010ef8566/issues/gh/airbytehq/airbyte/19898) we'll be wanting to send the full `jobOutput` there too, so I think we'll stop pulling those out in `execute` anyway